pkg/genyaml/testdata/multiline_comments/multiline_comments.yaml

```diff
@@ -0,0 +1,6 @@
+# StringField comment second line third line
+string1: string1
+
+
+# StringField comment \tsecond line \tthird line
```

Comment:
This doesn't seem like the desired behaviour here - I'm not sure that injecting \t into human-readable docs makes sense. Does GoDoc usually do this?

Reply:
The justification here is that `yaml.v3` supports a single [`HeadComment`](https://godoc.org/gopkg.in/yaml.v3#Node) so we have to cram the comment onto a single line. For `\t` and `\n`, specifically, this allows us to condense the extra whitespace. With that being said, I am not enamored of this behavior; what are your thoughts on making this more readable?

```diff
- # StringField comment 	 second line 	 third line
+ # StringField comment \t second line \t third line
```

Reply:
From the documentation it looks like `HeadComment` can maybe contain real newlines - could we just avoid collapsing it entirely?

As for the tabs, it would be really nice if we could just dedent it, but that might be slightly tricky given the first line sometimes lacks the indentation. I think it might make sense to just convert tabs to a space and subsequently coalesce multiple consecutive non-newline whitespace characters into a single space.

Reply:
I took your advice:
- [multiline comments are preserved](https://github.com/kubernetes/test-infra/pull/13637/commits/5b18160d003f4e0954ca7ebddbc10240f62f9633#diff-d75dc74a8d863cde946a6770c7c274b5R1-R4)
- [tabs are converted to spaces](https://github.com/kubernetes/test-infra/pull/13637/commits/5b18160d003f4e0954ca7ebddbc10240f62f9633#diff-d75dc74a8d863cde946a6770c7c274b5R27-R28)
- [multiple spaces are compressed into a single space](https://github.com/kubernetes/test-infra/pull/13637/commits/5b18160d003f4e0954ca7ebddbc10240f62f9633#diff-d75dc74a8d863cde946a6770c7c274b5R19-R20)